apis/kueue/v1alpha1/cohort_types.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	kueuebeta "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+)
+
+// CohortSpec defines the desired state of Cohort
+type CohortSpec struct {
+	// Parent references the name of the Cohort's parent, if
+	// any. It satisfies one of three cases:
```

Comment:
What about a cycle. What happens in that case?

Reply:
We disable all members of the Cohort graph. I updated documentation.